iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -393,28 +393,47 @@ public static PartitionData toPartitionData(StructLike sourceKey, Types.StructTy
     return data;
   }
 
-  public static List<DataFile> getDataFiles(Table table, int specId,
-      String partitionPath) {
+  /**
+   * Returns list of data files filtered by specId and partitionPath as following:
+   *  1. If matchBySpecId is true, then filters files by specId == file's specId, else by specId != file's specId
+   *  2. If partitionPath is not null, then also filters files where partitionPath == file's partition path
+   * @param table the iceberg table
+   * @param specId partition spec id
+   * @param partitionPath partition path
+   * @param matchBySpecId If true, searches for files matching the specId, else searches for ones that don't match
+   */
+  public static List<DataFile> getDataFiles(Table table, int specId, String partitionPath, boolean matchBySpecId) {
```

Comment:
wouldn't it be better to pass a filter
````
public static List<DataFile> getDataFiles(Table table, Predicate<DataFile> filter);
````
getDataFiles(table, file -> file.specId() == specId)
getDataFiles(table, file -> file.specId() != specId)

Reply:
Nice idea, done.